near_api/providers.py

```diff
@@ -56,7 +137,7 @@ def send_tx_and_wait(self, signed_tx: bytes, timeout: 'TimeoutType') -> dict:
                              timeout=timeout)
 
     def get_status(self, timeout: 'TimeoutType' = 2.0) -> dict:
-        r = requests.get("%s/status" % self.rpc_addr(), timeout=timeout)
+        r = self._session.get("%s/status" % self.rpc_addr(), timeout=timeout)
```

Comment:
This should go through the retry loop as well.  Currently this only retries on server errors but I don’t see why we wouldn’t want to retry on timeouts if we’re retrying other requests on timeout.

Reply:
Do you mean the same `TIMEOUT_ERROR` that is handled in `json_rpc`? Is it possible to get one here? In docs it among [`broadcast_tx_commit`](https://docs.near.org/api/rpc/transactions#send-transaction-await) errors and it's not mentioned among `/stats` endpoint errors (at least [here](https://docs.near.org/api/rpc/network#node-status)).

But yeah, since we'll be handling all timeouts ourselves then retry code might be the same and we can add handling of these error here as well :)

Reply:
Ah, I see.  It’s a different timeout that what I was thinking of.  What happens if requsets.get’s timeout runs out?

Also, at this point maybe we should move TIMEOUT_ERROR handling to _sign_and_submit_tx?  JsonProvider would handle 5xx and Account would handle TIMEOUT_ERROR and any other errors of this kind?

Reply:
> What happens if requsets.get’s timeout runs out?

I guess that requests's retry logic triggers and once number of attempts is exceeded an exception like requests.Timeout is thrown.

> Also, at this point maybe ...

Sounds great, I'll also be able to stick with built-in mechanisms for network/http error retries.